workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/container/BodyAndModalsScreen.kt

```diff
@@ -0,0 +1,20 @@
+package com.squareup.workflow1.ui.container
+
+import com.squareup.workflow1.ui.Screen
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+/**
+ * A screen that may stack a number of modal [Overlay]s over a body.
+ * While modals are present, the body is expected to ignore any
+ * input events -- touch, keyboard, etc.
+ */
```

Comment:
```suggestion
/**
 * A screen that may stack a number of modal [Overlay]s over a body.
 * While modals are present, the body is expected to ignore any
 * input events -- touch, keyboard, etc.
 * Passing an [Overlay] rendering as part of [modals] here marks that
 * Overlay as modal - that is, it should block input events around outside
 * of its content area. The logic for that input event blocking is handled
 * for users of this screen by the internal [BodyAndModalsContainer].
 */
```

Reply:
> it should block input events around outside of its content area

Oh, no no no no no. It should only block events within the bounds of the body. Blocking events outside of that area is a contract violation. And in fact turning on the "modal" flag for an Android Dialog implementation would be a bug precisely because of this.

I'm too tired to phrase this properly right now. Will come up with something before merging.

Reply:
Actually…let me leave this as is and rephrase with the next PR, where I'll actually be implementing proper custom dialogs. There may be more nuances then.